dags/aws.py

```diff
@@ -0,0 +1,30 @@
+import boto3
+from airflow import configuration
+
+ENDPOINT_URL = configuration.conf.get('libero', 'endpoint_url')
+
+
+def get_aws_connection(service: str):
+    """
+    Helper function created because endpoint_url setting (required for local
+    development) cannot be configured.
+    """
+    if ENDPOINT_URL:
+        return boto3.client(service, endpoint_url=ENDPOINT_URL)
+    else:
+        return boto3.client(service)
+
+
+def list_bucket_keys_iter(**list_objects_v2_params):
+    """
+    returns a generator that lists keys/prefixes in an AWS S3 bucket.
+    """
+    client = get_aws_connection('s3')
+    paginator = client.get_paginator('list_objects_v2')
+    for page in paginator.paginate(**list_objects_v2_params):
+        if 'Contents' in page:
+            for item in page['Contents']:
+                yield item['Key']
+        elif 'CommonPrefixes' in page:
```

Comment:
prefixes are akin to fake directories I think (if they end in `/`), what are they used for?

Reply:
Here I'm getting the top level "directory" as this should match the zip file name. I use this information to compare the incoming and expanded bucket to identify which zip files need to be worked on. This assumes that we want zipped files to be expanded in a directory matching the zip name.